dhcp/dhcp.go

```diff
@@ -0,0 +1,202 @@
+package dhcp
+
+import (
+	"fmt"
+	"html/template"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	cdconfig "github.com/coredhcp/coredhcp/config"
+	cdplugins "github.com/coredhcp/coredhcp/plugins"
+	cdserver "github.com/coredhcp/coredhcp/server"
+
+	pldns "github.com/coredhcp/coredhcp/plugins/dns"
+	plleasetime "github.com/coredhcp/coredhcp/plugins/leasetime"
+	plserverid "github.com/coredhcp/coredhcp/plugins/serverid"
+	plbootz "github.com/openconfig/bootz/dhcp/plugins/bootz"
+	plslease "github.com/openconfig/bootz/dhcp/plugins/slease"
+)
+
+const confTemplate = `
+# CoreDHCP configuration (yaml)
+server6:
+   plugins:
+     - server_id: LL {{ .IntfMacAddr }}
+     {{ if .BootzUrl }}
+     - bootz: {{ .BootzUrl }}
+     {{ end }}
+     {{ if .Dnsv6 }}
+     - dns: {{ .Dnsv6 }}
+     {{ end }}
+     {{ if .Ipv6Leases }}
+     - slease: {{ .Ipv6Leases }}
+     {{ end }}
+server4:
+  plugins:
+    - lease_time: 3600s
+    - server_id: {{ .IntfIpAddr }}
+    {{ if .BootzUrl }}
+    - bootz: {{ .BootzUrl }}
+    {{ end }}
+    {{ if .Dnsv4 }}
+    - dns: {{ .Dnsv4 }}
+    {{ end }}
+    {{ if .Ipv4Leases }}
+    - slease: {{ .Ipv4Leases }}
+    {{ end }}
+`
+
+var desiredPlugins = []*cdplugins.Plugin{
+	&plserverid.Plugin,
+	&plleasetime.Plugin,
+	&pldns.Plugin,
+	&plbootz.Plugin,
+	&plslease.Plugin,
+}
+
+// DHCPConfig contains the dhcp server configuration
+type DHCPConfig struct {
+	Interface  string
+	Dns        []string
+	AddressMap map[string]*DHCPEntry
+	BootzUrl   string
+}
+
+type DHCPEntry struct {
+	Ip string
+	Gw string
+}
+
+type dhcpServer struct {
+	server *cdserver.Servers
+}
+
+var instance *dhcpServer = nil
```

Comment:
It's basically to prohibit multiple instances of the server since all instances will bind to the same default port. If using other ports or having multiple instances is a use case then I can make the change.

Reply:
I see, thanks for clarifying.